earth_enterprise/src/fusion/autoingest/AssetSerializer.h

```diff
@@ -14,13 +18,81 @@ class AssetSerializerInterface {
     virtual ~AssetSerializerInterface() = default;
 };
 
+/*
+* GetFileInfo and ReadXMLDocument were added as function templates to allow
+* them to be overridden in unit tests. The versions here should be used for
+* all asset types during normal operation.
+*/
+template<class AssetType>
+bool GetFileInfo(const std::string &fname, uint64 &size, time_t &mtime) {
+  return khGetFileInfo(fname, size, mtime);
+}
+
+template<class AssetType>
+std::unique_ptr<GEDocument> ReadXMLDocument(const std::string &filename) {
+  return ReadDocument(filename);
+}
+
 template<class AssetType>
 class AssetSerializerLocalXML : public AssetSerializerInterface<AssetType>
 {
   public:
     virtual AssetPointerType<AssetType> Load(const std::string &boundref)
     {
-      return AssetType::Load(boundref);
+      std::string filename = AssetType::XMLFilename(boundref);
+      AssetPointerType<AssetType> result;
+      time_t timestamp = 0;
+      uint64 filesize = 0;
+
+      if (GetFileInfo<AssetType>(filename, filesize, timestamp) && (filesize > 0)) {
+        std::unique_ptr<GEDocument> doc = ReadXMLDocument<AssetType>(filename);
+        if (doc) {
+          try {
+            DOMElement *top = doc->getDocumentElement();
+            if (!top)
+              throw khException(kh::tr("No document element"));
+            std::string tagname = FromXMLStr(top->getTagName());
+            result = AssetFactory::CreateNewFromDOM<AssetType>(tagname,top);
+            if (nullptr == result) {
+              AssetThrowPolicy::WarnOrThrow
+                (kh::tr("Unknown asset type '%1' while parsing %2")
+                .arg(ToQString(tagname), ToQString(filename)));
+            }
+          } catch (const std::exception &e) {
+          AssetThrowPolicy::WarnOrThrow
```

Comment:
Indentation looks off here.

Reply:
Good catch. This code came from C++-inside-perl and the indentation was all off. That one slipped by.
There was also an inconsistency between the Load and Save methods here (2 vs 4 spaces). I'm making it all two-space indentation.